client/relay-chain-interface/src/lib.rs

```diff
@@ -110,6 +111,9 @@ pub trait RelayChainInterface: Send + Sync {
 	/// Get the hash of the current best block.
 	async fn best_block_hash(&self) -> RelayChainResult<PHash>;
 
+	/// Fetch the block header of a given height
+	async fn header(&self, block_id: BlockId) -> RelayChainResult<Option<PHeader>>;
```

Comment:
Please use `PHash`. No new `BlockId` in public interfaces anymore.

Reply:
But i specifically need to query using `BlockNumber`, any reason for this? 

> No new BlockId in public interfaces anymore

Reply:
Block numbers should only be used in places where there is no other way around them. Block numbers are not ambiguous. Some interfaces (like RPC/CLI) still require block numbers, but most "internal code" can be rewritten in a way to use hashes. E.g. get the header of block X and then go back using `parent_hash()`.

Reply:
> But i specifically need to query using BlockNumber, any reason for this?

The use case here is on-chain code which wishes to read the relay chain, they signal a request given a specific relay chain height (parachains don't have access to the relay chain blockhashes). So hence the `BlockId` here. 

Doesn't seem like there's any alternatives here

Reply:
And how does the parachain validates the relay chain headers?

Reply:
> The use case here is on-chain code which wishes to read the relay chain, they signal a request given a specific relay chain height (parachains don't have access to the relay chain blockhashes).

You say it signals which relay chain header it wants to read. So you are just wanting to get the header to get the storage root?

Reply:
yes

Reply:
Okay, when we got: https://github.com/paritytech/cumulus/issues/303

This can be changed.